### PR TITLE
New Profiles Migration

### PR DESCRIPTION
**Pull request by alexyoung23j**
## Link to Issue
Closes: #2982 

## Description of Changes
- Migrates the data from OffchainProfiles into Profiles as needed for the switch.
- The logic here is tricky to follow since the file is so long. I've added comments, but the basic flow is:
- For each Profile:
- - if they have set a profile_name
    - go through Addresses where profile_id is that profile.id. For each address, find the OffChainProfile that has that address_id. Choose the first OffchainProfile that has an avatarUrl as a field it the json blob in its “data” column, and add that avatarUrl to the avatar_url field on the Profile.
- - If no profile_name
    - If they have just one address
        - get the offchain profile for that address, copy in both the name and the avatarUrl into the new profile
    - If they have multiple addresses
        - if those addresses have same name
            - arbitrarily choose an avatar_url and name and put it in the profile
        - If they have different names
            - create a new user and profile for each of them with the corresponding avatar url and name.

## Test Plan
- Try this extensively locally and on staging. Non reversible change. 

## Deployment Plan

1. 

## Other Considerations

- The failing tests here are all just from the original bonnie.merge-profiles branch, none of this new work.

### CODE REVIEW

**Review by jnaviask — COMMENTED** [review]
A few comments that may help speed things up + question.

File: packages/commonwealth/server/migrations/20230307163345-profiles-migration.js [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+              {
+                type: queryInterface.sequelize.QueryTypes.SELECT,
+                transaction: t,
+              }
+            );
+
+            if (offchainProfiles.length === 1) {
+              let parsedData;
+              try {
+                parsedData = JSON.parse(offchainProfiles[0].data);
```
[the remark this answers is not in the document]
Comment (alexyoung23j) [reply packages/commonwealth/server/migrations/20230307163345-profiles-migration.js:74]: Hmm, this was per Zak's original ticket and OffchainProfile's name is the one that gets set in that "profile" modal in each community- not sure. @zakhap

File: packages/commonwealth/server/migrations/20230307163345-profiles-migration.js [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+          // User has set the profile_name via modal
+
+          const addresses = await queryInterface.sequelize.query(
+            `SELECT * FROM "Addresses" WHERE profile_id=${profile.id}`,
+            { type: queryInterface.sequelize.QueryTypes.SELECT, transaction: t }
+          );
+
+          for (const address of addresses) {
+            // Get offchainProfile for the address
+            const offchainProfiles = await queryInterface.sequelize.query(
```
[the remark this answers is not in the document]
Comment (alexyoung23j) [reply packages/commonwealth/server/migrations/20230307163345-profiles-migration.js]: good call

**Review by zakhap — COMMENTED** [review]
Closing this branch+PR, moving Migration to a separate PR. Great work @alexyoung23j

**Pull request closed without merging**